src/local/update.go

```diff
@@ -9,46 +9,44 @@ import (
 func (m *Manager) UpdatePaths(state model.ObjectState, rename bool) {
 	object := state.LocalOrRemoteState()
 
-	switch v := state.(type) {
-	case *model.BranchState:
-		branch := object.(*model.Branch)
-
-		// Set paths
-		if v.Path == "" || rename {
-			v.Path = m.Naming().BranchPath(branch, branch.IsDefault)
-		}
-		v.ResolveParentPath()
-	case *model.ConfigState:
-		config := object.(*model.Config)
-
-		// Get parent - branch
-		branchKey := v.BranchKey()
-		branchManifest, found := m.manifest.GetRecord(branchKey)
-		if !found {
-			panic(fmt.Errorf("branch manifest wit key \"%s\" not found", branchKey))
+	// Update parent path
+	m.manifest.ResolveParentPath(state.Manifest())
+
+	// Re-generate object path IF rename is enabled OR path is not set
+	if state.GetObjectPath() == "" || rename {
+		switch v := state.(type) {
+		case *model.BranchState:
+			v.PathInProject = m.Naming().BranchPath(object.(*model.Branch))
+		case *model.ConfigState:
+			config := object.(*model.Config)
+			v.PathInProject = m.Naming().ConfigPath(v.ParentPath, v.Component, config)
+		case *model.ConfigRowState:
+			row := object.(*model.ConfigRow)
+			v.PathInProject = m.Naming().ConfigRowPath(v.ParentPath, row)
+		default:
+			panic(fmt.Errorf(`unexpect type "%T"`, state))
 		}
+	}
+}
 
-		// Set paths
-		if v.Path == "" || rename {
-			v.Path = m.Naming().ConfigPath(v.Component, config)
-		}
-		v.ResolveParentPath(branchManifest.(*model.BranchManifest))
-	case *model.ConfigRowState:
-		row := object.(*model.ConfigRow)
-
-		// Get parent - config
-		configKey := row.ConfigKey()
-		configManifest, found := m.manifest.GetRecord(configKey)
-		if !found {
-			panic(fmt.Errorf("config manifest wit key \"%s\" not found", configKey))
-		}
+func (m *Manager) UpdateBlockPath(block *model.Block, rename bool) {
```

Comment:
Rovno som tu presunul aj premenovani `block/code` z transformacii, nech je to spolu.